lib/core/src/Cardano/Wallet/Api.hs

```diff
@@ -146,6 +148,11 @@ type ListTransactions t = "wallets"
     :> QueryParam "order" (ApiT SortOrder)
     :> Get '[JSON] [ApiTransaction t]
 
+-- | https://input-output-hk.github.io/cardano-wallet/api/#operation/postExternalTransaction
+type PostExternalTransaction t = "externalTransactions"
```

Comment:
1. As a user of the API I would prefer to post the transaction data directly with `content-type: application/octet-stream`. Rather than having to base64-encode it then stuff it into a json object. Maybe the API could support both types of request bodies if json is absolutely required.

2. And it looks like we can't give correct/useful answers for many of the fields of ApiTransaction. What about a specialised response type which just returns the txid?

Reply:
agree 100% to the point 2. I will respond with TxId here

When it comes to `content-type: application/octet-stream` I will support it along with json in the next PR concluding this story if you are ok with this (this PR got really big)